modules/dhcp/providers/server/infoblox.rb

```diff
@@ -0,0 +1,145 @@
+require 'dhcp/subnet'
+require 'dhcp/record'
+require 'dhcp/record/reservation'
+require 'dhcp/server'
+require 'infoblox'
+
+module Proxy::DHCP
+  # Represents Infoblox DHCP Server (https://www.infoblox.com) uses https://github.com/govdelivery/infoblox >= 0.4.1
+  class Infoblox < Server
+
+    def initialize(options = {})
+      super options[:server]
+      @username = options[:username]
+      @password = options[:password]
+    end
+
+   def delRecord subnet, record
+      validate_subnet subnet
+      validate_record record
+      # TODO: Refactor this into the base class
+      raise InvalidRecord, "#{record} is static - unable to delete" unless record.deleteable?
+      # "Deleting"" a record here means just disabling dhcp
+      connection = ::Infoblox::Connection.new(:username => Proxy::DhcpPlugin.settings.username, :password => Proxy::DhcpPlugin.settings.password, :host => Proxy::DhcpPlugin.settings.dhcp_server)
```

Comment:
You might consider creating a `connection` method... this line is repeated several times in different methods.